File: src/main/java/com/wgzhao/addax/admin/dto/TaskResultDto.java
package com.wgzhao.addax.admin.dto;


public record TaskResultDto(boolean success, String message, long durationSeconds)
{
    public static TaskResultDto success(String message, long durationSeconds) {
        return new TaskResultDto(true, message, durationSeconds);
    }

    public static TaskResultDto failure(String message, long durationSeconds) {
        return new TaskResultDto(false, message, durationSeconds);
    }
}

File: src/main/kotlin/com/wgzhao/addax/admin/dto/TaskResultDto.kt
package com.wgzhao.addax.admin.dto

data class TaskResultDto(
    val success: Boolean = false,
    val message: String? = null,
    val durationSeconds: Long = 0
) {
    companion object {
        @JvmStatic
        fun success(message: String, durationSeconds: Long): TaskResultDto {
            return TaskResultDto(true, message, durationSeconds)
        }

        @JvmStatic
        fun failure(message: String, durationSeconds: Long): TaskResultDto {
            return TaskResultDto(false, message, durationSeconds)
        }
    }
}
